File: ProjektGrupowy/Scripts/App/Mobile/ViewControllers/HomePageController.js
/// <reference path="../../_references.js" />

ProjektGrupowy.App.Mobile.viewControllers.HomePage = (function () {

    var c = new ProjektGrupowy.Framework.classes.Controller('Mobile', 'HomePage');

    return {

        actions: {

            loginForm: function () {
                return c.action('LoginForm', {});
            },

            registerForm: function () {
                return c.action('RegisterForm', {});
            },

            joinGameForm: function () {
                return c.action('JoinGameForm', {});
            }
        }
    }

}());

File: ProjektGrupowy/Areas/Site/Controllers/HomePageController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Web.Hosting;
using System.Threading;
using ProjektGrupowy.Models.Platform;
using ProjektGrupowy.Models.Database.DAO;
using ProjektGrupowy.Models.Game.Common;
using ProjektGrupowy.Models.Core;

namespace ProjektGrupowy.Areas.Site.Controllers
{
    public class HomePageController : Controller
    {
        public ActionResult Index()
        {
            if (Request.Browser.IsMobileDevice)
            {
                return RedirectToAction("Index", "HomePage", new { area = "Mobile" });
            }
            Platform.OnEnterSite();
            return View("HomePage");
        }

        public ActionResult LoginForm()
        {
            return View("Forms/LoginForm");
        }

        public ActionResult JoinGameForm()
        {
            return View("Forms/JoinGameForm");
        }

        public ActionResult RegisterForm()
        {
            return View("Forms/RegisterForm");
        }

        public ActionResult NewGameForm()
        {
            return View("Forms/NewGameForm");
        }

    }
}